File: auto-auction-frontend/src/API/LotService.js
import axios from "axios";
const API_URL = "http://localhost:8080/api/lots/";

export default class LotService {
  static async getLotById(id) {
    const response = await axios.get(API_URL + `getLotById/${id}`);
    return response;
  }

  static async getLots() {
    const response = await axios.get(API_URL + "getLots");
    return response;
  }

  static async getNewLots() {
    const response = await axios.get(API_URL + "getNewLots");
    return response;
  }
}

File: auto-auction-backend/src/main/java/com/valery/services/LotService.java
package com.valery.services;

import com.valery.entities.Lot;
import com.valery.exceptions.NotFoundException;

public interface LotService {
	Lot getLotById(Long id) throws NotFoundException;

	Iterable<Lot> getLots();

	Iterable<Lot> getNewLots();
}
